s32/soc/s32k344/include/Emios_Icu_Ip_SA_Init_PBcfg.h

```diff
@@ -68,16 +68,918 @@ extern "C"{
 ==================================================================================================*/
 #if (STD_ON == EMIOS_ICU_IP_USED)
 
+#if DT_REG_ADDR(DT_INST(0, nxp_s32_emios)) == IP_EMIOS_0_BASE
+#define EMIOS_0_DRV_INST    0
+#elif DT_REG_ADDR(DT_INST(1, nxp_s32_emios)) == IP_EMIOS_0_BASE
+#define EMIOS_0_DRV_INST    1
+#else
+#define EMIOS_0_DRV_INST    2
+#endif
 
-#ifndef ICU_EMIOS_0_CH_0_ISR_USED
-    /** @brief Macros for indicate EMIOS interrupts used by ICU. */
-    #define ICU_EMIOS_0_CH_0_ISR_USED    (STD_ON)
+#if DT_REG_ADDR(DT_INST(0, nxp_s32_emios)) == IP_EMIOS_1_BASE
+#define EMIOS_1_DRV_INST    0
+#elif DT_REG_ADDR(DT_INST(1, nxp_s32_emios)) == IP_EMIOS_1_BASE
+#define EMIOS_1_DRV_INST    1
+#else
+#define EMIOS_1_DRV_INST    2
+#endif
+
+#if DT_REG_ADDR(DT_INST(0, nxp_s32_emios)) == IP_EMIOS_2_BASE
+#define EMIOS_2_DRV_INST    0
+#elif DT_REG_ADDR(DT_INST(1, nxp_s32_emios)) == IP_EMIOS_2_BASE
+#define EMIOS_2_DRV_INST    1
+#else
+#define EMIOS_2_DRV_INST    2
 #endif
 
+#define EMIOS_PWM_NODE(n)   DT_CHILD(DT_INST(n, nxp_s32_emios), pwm)
+
+#define EMIOS_ICU_CHANNEL(node_id, ch)                                            \
+        IF_ENABLED(DT_ENUM_HAS_VALUE(node_id, pwm_mode, SAIC),                    \
+                  ((DT_PROP(node_id, channel) == ch) ||                           \
+                    IF_ENABLED(DT_NODE_HAS_PROP(node_id, master_bus),             \
+                    (DT_PROP_BY_PHANDLE(node_id, master_bus, channel) == ch) ||)  \
+                  ))
+
+/*---------------------------------------EMIOS 0-------------------------------------------*/
+#if DT_NODE_HAS_STATUS(EMIOS_PWM_NODE(EMIOS_0_DRV_INST), okay)
+
+#if DT_FOREACH_CHILD_STATUS_OKAY_VARGS(EMIOS_PWM_NODE(EMIOS_0_DRV_INST), \
+                                       EMIOS_ICU_CHANNEL, 0) 0
 #ifndef EMIOS_0_CH_0_ISR_USED
-    /** @brief Macros for indicate EMIOS interrupts used. */
-    #define EMIOS_0_CH_0_ISR_USED        (STD_ON)
+#define EMIOS_0_CH_0_ISR_USED        (STD_ON)
```

Comment:
is it possible to generate the configuration for all emios instances and channels, so that we reduce the number of patches needed over the generated code?

Reply:
I've update with generated code for all instances and channel